source/initial_conditions/interface.cc

```diff
@@ -85,68 +91,127 @@ namespace aspect
+          initial_temperature_objects.back()->initialize ();
+        }
     }
 
 
+
     template <int dim>
-    Interface<dim> *
-    create_initial_conditions(const std::string &model_name)
+    double
+    Manager<dim>::initial_temperature (const Point<dim> &position) const
     {
-      Interface<dim> *plugin = std_cxx11::get<dim>(registered_plugins).create_plugin (model_name,
-                                                                                      "Initial conditions::Model name");
-      return plugin;
+      double temperature = 0.0;
+      for (typename std::list<std_cxx11::shared_ptr<InitialConditions::Interface<dim> > >::const_iterator
+           initial_temperature_object = initial_temperature_objects.begin();
+           initial_temperature_object != initial_temperature_objects.end();
+           ++initial_temperature_object)
+        {
+          temperature += (*initial_temperature_object)->initial_temperature(position);
+        }
+      return temperature;
     }
 
 
     template <int dim>
-    std::string
-    get_valid_model_names_pattern ()
+    const std::vector<std::string> &
+    Manager<dim>::get_active_initial_temperature_names () const
     {
-      return std_cxx11::get<dim>(registered_plugins).get_pattern_of_names ();
+      return model_names;
     }
 
 
+    template <int dim>
+    const std::list<std_cxx11::shared_ptr<Interface<dim> > > &
+    Manager<dim>::get_active_initial_temperature_conditions () const
+    {
+      return initial_temperature_objects;
+    }
+
 
     template <int dim>
     void
-    declare_parameters (ParameterHandler &prm)
+    Manager<dim>::declare_parameters (ParameterHandler &prm)
     {
-      // declare the entry in the parameter file
+      // declare the actual entry in the parameter file
       prm.enter_subsection ("Initial conditions");
       {
         const std::string pattern_of_names
-          = get_valid_model_names_pattern<dim> ();
+          = std_cxx11::get<dim>(registered_plugins).get_pattern_of_names ();
+
+        prm.declare_entry("List of model names",
+                          "",
+                          Patterns::MultipleSelection(pattern_of_names),
+                          "A comma separated list of initial temperature models that "
+                          "will be used to initialize the temperature. "
+                          "The results of each of these models will be added.\n\n"
+                          "The following initial tempeature models are available:\n\n"
+                          +
+                          std_cxx11::get<dim>(registered_plugins).get_description_string());
+
         prm.declare_entry ("Model name", "unspecified",
                            Patterns::Selection (pattern_of_names+"|unspecified"),
                            "Select one of the following models:\n\n"
+                           "Warning: This is the old formulation of specifying "
```

Comment:
I think you want to move these three new lines below the list of descriptions -- the sentence structure doesn't work right now.

This is latex, so you can do `\\textbf{Warning:}`. Also, it's not entirely clear what "This" refers to here -- say "Warning: This parameter provides an old and deprecated way for specifying initial temperature models. Use 'List of model names' instead."